src/components/Catalyst/Step5.js

```diff
@@ -0,0 +1,235 @@
+// @flow
+
+/**
+ * Step 5 for the Catalyst registration
+ * Ask password used for signing transaction
+ * submit the transaction
+ * ### HW is NOT supported yet - validation is done on first screen itself###
+ */
+
+import _ from 'lodash'
+import React, {useEffect, useState} from 'react'
+import {View, SafeAreaView} from 'react-native'
+import {injectIntl, defineMessages} from 'react-intl'
+import {connect} from 'react-redux'
+
+import type {ComponentType} from 'react'
+import type {IntlShape} from 'react-intl'
+
+import {CONFIG} from '../../config/config'
+import KeyStore from '../../crypto/KeyStore'
+import {formatTokenWithSymbol} from '../../utils/format'
+import {showErrorDialog, submitTransaction} from '../../actions'
+import {
+  Text,
+  ProgressStep,
+  Button,
+  OfflineBanner,
+  ValidatedTextInput,
+  StatusBar,
+  PleaseWaitModal,
+} from '../UiKit'
+import ErrorModal from '../Common/ErrorModal'
+import {withTitle} from '../../utils/renderUtils'
+import {
+  CATALYST_ROUTES,
+  SEND_ROUTES,
+  WALLET_ROOT_ROUTES,
+} from '../../RoutesList'
+import walletManager, {SystemAuthDisabled} from '../../crypto/walletManager'
+import globalMessages, {
+  errorMessages,
+  confirmationMessages,
+  txLabels,
+} from '../../i18n/global-messages'
+import {WrongPassword} from '../../crypto/errors'
+
+import styles from './styles/Step5.style'
+
+import {
+  easyConfirmationSelector,
+  defaultNetworkAssetSelector,
+} from '../../selectors'
+import type {Navigation} from '../../types/navigation'
+
+const messages = defineMessages({
+  subTitle: {
+    id: 'components.catalyst.step5.subTitle',
+    defaultMessage: '!!!Confirm Registration',
+  },
+  description: {
+    id: 'components.catalyst.step5.description',
+    defaultMessage:
+      '!!!Enter the spending password to confirm voting registration and submit the certificate generated in previous step to blockchain',
+  },
+})
+
+const Step5 = ({
+  intl,
+  isEasyConfirmationEnabled,
+  navigation,
+  submitTransaction,
+  unSignedTx,
+  defaultAsset,
+}) => {
+  const [password, setPassword] = useState(
+    CONFIG.DEBUG.PREFILL_FORMS ? CONFIG.DEBUG.PASSWORD : '',
+  )
+
+  const [sendingTransaction, setSendingTransaction] = useState(false)
+  const [errorData, setErrorData] = useState({
+    showErrorDialog: false,
+    errorMessage: '',
+    errorLogs: '',
+  })
+  const [fees, setFees] = useState(null)
+
+  useEffect(() => {
+    unSignedTx.fee().then((o) => {
+      setFees(o.getDefault())
+    })
+  }, [])
+
+  const isConfirmationDisabled = !isEasyConfirmationEnabled && !password
+
+  const submitTrx = async (decryptedKey) => {
+    setSendingTransaction(true)
+    await submitTransaction(unSignedTx, decryptedKey)
```

Comment:
if this call fails we can get stuck with `sendingTransaction = true`. Usually it's wrapped in a try-finally.

Reply:
its handed where `submitTx` is called. But I'm now adding trycatch with finally block